### ISSUE TITLE
extrinsic_mat is not a file in the archive (Stereo setup - off-axis)

### ISSUE THREAD

**Issue by tahasamavati** [Cartucho/vision_blender#18]
Hello,

When trying to get the extrinsic matrix corresponding to the non-pivot camera (in my case the right one). It says 'extrinsic_mat is not a file in the archive '. My setup is off-axis. It seems that it just works for a parallel stereo setup.

**Comment by Cartucho** [comment]
Yes, I don't really know how to get extrinsic of an `off-axis` camera, since I don't really know what `off-axis` means. I was not able to find a way to get the world transform of the non-pivot camera, so I hand-coded the parallel transformation using the baseline. I could do the same to the off-axis if I understand what it means.

Also for the disparity calculation, we need the images to be rectified, which in practice would mean that image planes are parallel. So probably for that case I wouldn't be able to generate disparity maps directly without having to rectify the images (which I would do using OpenCV).

**Comment by Cartucho** [comment]
I will google a bit more to see if we can get the world-coordinates of the non-pivot camera directly

**Comment by Cartucho** [comment]
Hi! So I made [some progress](https://blender.stackexchange.com/questions/233877/how-to-calculate-the-transformation-from-the-pivot-point-to-each-stereo-camera) on how to get the extrinsic matrix for both cameras but I am still confused about how those models work.

Anyway, the add-on should still work even when we do not know the extrinsic. It should simply send a warning to the user saying that the disparity can only be calculated, for now, in the case of parallel stereo. I will push a commit so that the add-on keeps functioning and shows the warning!

**Comment by tahasamavati** [comment]
Hi, I switched to the parallel mode for now. But to help solve this issue, I googled a bit and did not find any kind of geometric model for the transformation between cameras in this mode. If I find something useful, I will comment here.
This [link ](http://paulbourke.net/stereographics/stereorender/) describes the off-axis mode but there is no geometric model provided.
